src/MiniExcel/SaveByTemplate/InputValueExtractor.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Reflection;
+using MiniExcelLibs.Utils;
+
+namespace MiniExcelLibs.OpenXml.SaveByTemplate
+{
+    public static class InputValueExtractor
+    {
+        public static IDictionary<string, object> ToValueDictionary(object valueObject)
+            => valueObject is Dictionary<string, object> valueDictionary
+                ? GetValuesFromDictionary(valueDictionary)
+                : GetValuesFromObject(valueObject);
+
+        private static IDictionary<string, object> GetValuesFromDictionary(Dictionary<string, object> valueDictionary)
+        {
+            return valueDictionary
+                .ToDictionary(
+                    x => x.Key,
+                    x => x.Value is IDataReader dataReader
+                        ? TypeHelper.ConvertToEnumerableDictionary(dataReader).ToList()
+                        : x.Value);
+        }
+
+        private static IDictionary<string, object> GetValuesFromObject(object valueObject)
+        {
+            var type = valueObject.GetType();
+
+            var propertyValues = type
+                .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                .Select(property => new { property.Name, Value = property.GetValue(valueObject) });
+
+            var fieldValues = type
+                .GetFields(BindingFlags.Public | BindingFlags.Instance)
+                .Select(field => new { field.Name, Value = field.GetValue(valueObject) });
```

Comment:
The old code had a check that a field hasn't been previously added as a property. I removed this check because I can't think how this can be ever true. But please correct me if I am wrong.